llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -464,11 +481,34 @@ bool AIEHazardRecognizer::checkConflict(
     int DeltaCycles, std::optional<int> FUDepthLimit) {
   assert(Scoreboard.isValidDelta(DeltaCycles));
 
+  if (checkFormatConflict(Scoreboard, DeltaCycles, SlotSet))
```

Comment:
I was wondering if we could specify conflict types as bits in a bit sequence/unsigned, instead of boolean. I say it because we almost reimplement this method entirely in `onlyFormatHazard`. I was thinking in something like:

```
if (Conflicts::Format == checkConflict(...) ) {

}

```

In this way we can query any individual conflict without a custom implementation...

Reply:
Or `checkConflict` can be a wrapper around this function that can give all conflicts as a bitmask....

Reply:
I think splitting into multiple `checkXXXXConclict` is great! From what I see in the code it seems to be enough to be able to query specific conflicts, and nobody really checks the bitset. Could we keep the original `checkConflict` as returning a `bool`, or am I missing something?